docker/beam-environment/entrypoint.sh

```diff
@@ -0,0 +1,362 @@
+  cd "$BEAM_NAME" || echo "ERROR: The dir '$BEAM_NAME' is not available"
+
+  if [[ $BEAM_COMMIT_SHA ]]; then
+    echo "Resetting the branch '$BEAM_BRANCH_NAME' to commit '$BEAM_COMMIT_SHA'"
+    git reset --hard "$BEAM_COMMIT_SHA";
+  else
+    BEAM_COMMIT_SHA=$(git log -1 --pretty=format:%H)
+    echo "Using the latest commit in the branch '$BEAM_BRANCH_NAME' - '$BEAM_COMMIT_SHA'"
+  fi
+else
+  echo "Pulling the code is disabled (PULL_CODE set to '$PULL_CODE'), using code mounted to '$PATH_TO_PROJECT_PARENT'."
+  BEAM_COMMIT_SHA=$(git log -1 --pretty=format:%H)
+  BEAM_BRANCH_NAME=$(git symbolic-ref --short HEAD)
+  echo "Using the latest commit in the branch '$BEAM_BRANCH_NAME' - '$BEAM_COMMIT_SHA'"
+fi
+
+
+##
+## remember the BEAM working folder location
+##
+BEAM_PATH=$(pwd)
+echo "Working from '$BEAM_PATH'"
+
+
+##
+## Required to avoid configuring ssh keys because by-default all our git data submodules configured to use ssh.
+##
+if [ "$ENFORCE_HTTPS_FOR_DATA_REPOSITORY" = true ]; then
+  echo "Forcing git to use https over ssh\git url. In order to pull public repository without configuring ssh keys."
+  git config --global url."https://github.com/LBNL-UCB-STI".insteadOf "git@github.com:LBNL-UCB-STI"
+fi
+
+
+##
+## pulling data from github if enabled or checking if data location was mounted separately
+##
+if [ "$PULL_DATA" = true ]; then
+  echo "Pulling the data from github (PULL_DATA set to '$PULL_DATA')."
+  production_data_submodules=$(git submodule | awk '{ print $2 }')
+  for i in $production_data_submodules;
+    do
+      if [[ $BEAM_CONFIG  == $i* ]];
+      then
+        echo "Loading remote production data for $i"
+        git config "submodule.$i.branch" "$BEAM_DATA_BRANCH_NAME"
+        git submodule update --init --remote "$i"
+        cd "$i" || echo "ERROR: The path '$i' is not available"
+        if [[ $BEAM_DATA_COMMIT_SHA ]]; then
+          echo "Checking out the data commit '$BEAM_DATA_COMMIT_SHA'"
+          git checkout "$BEAM_DATA_COMMIT_SHA"
+        else
+          BEAM_DATA_COMMIT_SHA=$(git log -1 --pretty=format:%H)
+          echo "Latest commit is '$BEAM_DATA_COMMIT_SHA'"
+        fi
+        cd - || echo "ERROR: Can't move to the previous location"
+      fi;
+    done
+
+  echo "Doing lfs pull"
+  git lfs pull
+else
+  echo "Pulling the data from github is disabled (PULL_DATA set to '$PULL_DATA')."
+  COMBINED_CONFIG_PATH="/app/data/${BEAM_CONFIG#*/}"
+  echo "Trying combined path from data volume: '$COMBINED_CONFIG_PATH'"
+  if [ -e "$COMBINED_CONFIG_PATH" ]; then
+    BEAM_CONFIG=$COMBINED_CONFIG_PATH
+    echo "File exist, using config from mounted data volume - '$BEAM_CONFIG'"
+  else
+    echo "File does not exist, going to use data from inside beam folder."
+  fi
+fi
+
+
+##
+## notification
+##
+if [ "$SEND_NOTIFICATION" = true ]; then
+  send_slack_notification "Run Started
+    Run Name $TITLED
+    Instance ID $INSTANCE_ID
```

Comment:
fixed

Reply:
There are still notification items that are probably not useful, but they are in the optional field, so I guess it would just be empty - and that is probably fine